code/game/objects/structures/wallmounts/oxygen.dm

```diff
@@ -0,0 +1,35 @@
+/obj/structure/wall_cabinet/oxygen
+	name = "oxygen cabinet"
+	desc = "A small wall mounted cabinet designed to hold a breath mask and an emergency oxygen tank."
+	icon = 'icons/obj/wallmounts.dmi'
+	icon_state = "oxygen_closed"
+	nick = "oxygen"
+	capacity = 2
+	exclusivity = 1
+	allowedContent = list(/obj/item/weapon/tank/internals/emergency_oxygen, /obj/item/clothing/mask/breath)
+	spawnContent = newlist(/obj/item/weapon/tank/internals/emergency_oxygen, /obj/item/clothing/mask/breath)
+
+
+/obj/structure/wall_cabinet/oxygen/New(loc, ndir, building)			// This is a child of wall cabinet
+	..()															// Please look at /obj/structure/wall_cabinet to understand this code
+
+
+/obj/structure/wall_cabinet/oxygen/update_icon()					// Overrides update_icon() because of special naming system requiring more different states
+	cut_overlays()													// Files are named "oxygen_tank" and "oxygen_mask" for overlays
+	if(!opened)
+		icon_state = "oxygen_closed"
```

Comment:
you can make all wall cabinet icon states formated the same and set their icon state to 
`[initial(icon_state)]_closed`
`[initial(icon_state)]_open`
 in the base wall cabinet. Have that parent proc return 1 if closed or 0 if open and children procs can do 
`if(..())  if(findContent(new/obj/item/weapon/tank/internals/emergency_oxygen))`